torch/csrc/autograd/python_engine.cpp

```diff
@@ -3,13 +3,33 @@
 #include "torch/csrc/autograd/engine.h"
 #include "torch/csrc/THP.h"
 #include "torch/csrc/DynamicTypes.h"
+#include "torch/csrc/utils/auto_gil.h"
 
 using namespace torch::autograd;
 
 struct THPEngine {
     PyObject_HEAD
 };
 
+struct PythonEngine : public Engine {
+  virtual void thread_main(ReadyQueue& queue) override {
+    AutoGIL gil;
+    AutoNoGIL no_gil;
```

Comment:
Is the point of these two lines that, this new threads needs to have a thread state associated because some Python function may be called inside (somewhere during the backward), but we actually don't want to hold the GIL right now and the GIL will be locked just before calling the python function later?